### PR TITLE
Test URL Filtering from the command line

### PR DESCRIPTION
**Pull request by jnioche**
Implements #1079

For instance

`storm local target/xxx.jar  com.digitalpebble.stormcrawler.filtering.URLFilters https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed | grep -v main`

to test the filtering pipeline present in the jar.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,6 +34,23 @@ The [WIKI](https://github.com/DigitalPebble/storm-crawler/wiki) is a good place
...
+mvn git-code-format:format-code -Dgcf.globPattern=**/*
+```
+
+Each commit must include a DCO which looks like this
+
+```
+Signed-off-by: Jane Smith <jane.smith@email.com>
+```
+
+You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `– – signoff` to add the `Signed-off-by` line to the end of the commit message.
```
Comment (rzo1) [inline README.md:51]: `user.name` and `user.email`

Comment (rzo1) [reply README.md:51]: git configs -> git config

File: core/src/main/java/com/digitalpebble/stormcrawler/filtering/URLFilters.java [hunk 2]
```diff
@@ -49,7 +56,7 @@ private URLFilters() {
         filters = new URLFilters[0];
     }
 
-    private String configFile = "urlfilters.config.file";
+    private String configFile = "urlfilters.json";
```
Comment (rzo1) [inline core/src/main/java/com/digitalpebble/stormcrawler/filtering/URLFilters.java:59]: Is there a reason for actually changing this value? Was it a bug? :)

**Review by rzo1 — APPROVED** [review]

File: core/src/main/java/com/digitalpebble/stormcrawler/filtering/URLFilters.java (hunk 2 above)
Comment (jnioche) [reply core/src/main/java/com/digitalpebble/stormcrawler/filtering/URLFilters.java:59]: in a way yes, it was a confusion between the key name for loading from the conf with the placeholder actual value. Didn't matter much as the default value was always overridden but at least now it is correct and in a way documents what the default value is

**Comment by jnioche** [comment]
@rzo1 I couldn't get the -s option to work for some reason. Any chance you could have a look at it?

**Comment by rzo1** [comment]
I am running this in Intelli: 

```bash
"https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed" -s "https://pubmed.ncbi.nlm.nih.gov/18926286/"
```

and the `-s` option is parsed. For my tests with

```json
{
	"com.digitalpebble.stormcrawler.filtering.URLFilters": [
		{
			"class": "com.digitalpebble.stormcrawler.filtering.basic.BasicURLFilter",
			"name": "BasicURLFilter",
			"params": {
				"maxPathRepetition": 3,
				"maxLength": 1024
			}
		},
		{
			"class": "com.digitalpebble.stormcrawler.filtering.depth.MaxDepthFilter",
			"name": "MaxDepthFilter",
			"params": {
				"maxDepth": -1
			}
		},
		{
			"class": "com.digitalpebble.stormcrawler.filtering.basic.BasicURLNormalizer",
			"name": "BasicURLNormalizer",
			"params": {
				"removeAnchorPart": true,
				"unmangleQueryString": true,
				"checkValidURI": true,
				"removeHashes": true,
				"hostIDNtoASCII": true
			}
		},
		{
			"class": "com.digitalpebble.stormcrawler.filtering.host.HostURLFilter",
			"name": "HostURLFilter",
			"params": {
				"ignoreOutsideHost": false,
				"ignoreOutsideDomain": true
			}
		},
		{
			"class": "com.digitalpebble.stormcrawler.filtering.basic.SelfURLFilter",
			"name": "SelfURLFilter"
		},
		{
			"class": "com.digitalpebble.stormcrawler.filtering.sitemap.SitemapFilter",
			"name": "SitemapFilter"
		}
	]
}

```

it outputs:

```bash
	[com.digitalpebble.stormcrawler.filtering.basic.BasicURLFilter] 1786msec => https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed
	[com.digitalpebble.stormcrawler.filtering.depth.MaxDepthFilter] 0msec => https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed
	[com.digitalpebble.stormcrawler.filtering.basic.BasicURLNormalizer] 118msec => https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed
	[com.digitalpebble.stormcrawler.filtering.host.HostURLFilter] 1889msec => https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed
	[com.digitalpebble.stormcrawler.filtering.basic.SelfURLFilter] 0msec => https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed
	[com.digitalpebble.stormcrawler.filtering.sitemap.SitemapFilter] 0msec => https://pubmed.ncbi.nlm.nih.gov/18926286/?format=pubmed
```

**Comment by rzo1** [comment]
Next test will be via storm cli :-) (most likely in the next days)

**Comment by rzo1** [comment]
Looks like Storm cli messes around with CLI options (as it allows them as well). Maybe we better switch to a custom format like key=val

**Comment by jnioche** [comment]
> Looks like Storm cli messes around with CLI options (as it allows them as well). Maybe we better switch to a custom format like key=val

I went for the simpler option of just passing the source as an optional 2nd argument. works for now

**Comment by jnioche** [comment]
thanks for the review @rzo1

**Pull request merged**